### PR TITLE
docs: clarifies MySQL catalog migration process

### PR DESCRIPTION
**Pull request by joergsteffens**
Also:
split Howtos.rst into one file per section
and some cleanups.

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

##### General

- [x] PR name is meaningful
- [x] Purpose of the PR is understood
- [x] Separate commit for this PR in the CHANGELOG.md, PR number referenced is same
- [x] Commit descriptions are understandable and well formatted

##### Source code quality

- ~~[ ] Source code changes are understandable~~
- ~~[ ] Variable and function names are meaningful~~
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR
- [x] `bareos-check-sources --since-merge` does not report any problems
- [x] `git status` should not report modifications in the source tree after building and testing

### CODE REVIEW

File: docs/manuals/source/Appendix/Howtos/RemovingAClient.rst.inc [hunk 1]
```diff
@@ -0,0 +1,37 @@
+.. _section-RemoveClient:
```
Comment (bruno-at-bareos) [inline docs/manuals/source/Appendix/Howtos/RemovingAClient.rst.inc:1]: Shouldn't we use the same name for the file RemoveClient.rst.inc (the use of AClient doesn't seems to bring any useful information, even can add confusion in reading)

File: docs/manuals/source/Appendix/Howtos/StatisticsCollection.rst.inc [hunk 2]
```diff
@@ -0,0 +1,36 @@
+.. _section-StatisticCollection:
```
Comment (bruno-at-bareos) [inline docs/manuals/source/Appendix/Howtos/StatisticsCollection.rst.inc:1]: StatisticsCollection.rst.inc versus StatisticCollection section name: seems logical to fix one of them

File: docs/manuals/source/IntroductionAndTutorial/InstallingBareos.rst [hunk 3]
```diff
@@ -317,8 +317,7 @@ If you are not interested in this additional functionality, the commands describ
 Prepare Bareos database
 -----------------------
 
-We assume that you have already your database installed and basically running.
-Using the PostgreSQL database backend is recommended.
+We assume that you already have your database server installed and basically running.
```
Comment (bruno-at-bareos) [inline docs/manuals/source/IntroductionAndTutorial/InstallingBareos.rst:320]:
```
We assume that you already have your PostgreSQL database server installed and basically running.
```
So basically we remember any reader that only PostgreSQL is actually supported. (Perhaps he got previous informations about the support of different backend)

File: docs/manuals/source/IntroductionAndTutorial/UpdatingBareos.rst [hunk 4]
```diff
@@ -3,7 +3,8 @@
 Updating Bareos
 ===============
 
-In most cases, a Bareos update is simply done by a package update of the distribution. Please remind, that Bareos Director and Bareos Storage Daemon must always have the same version. The version of the File Daemon may differ, see chapter about :ref:`backward compatibility <backward-compatibility>`.
+In most cases, a Bareos update is simply done by a package update of the distribution.
+Please remind, that Bareos Director and Bareos Storage Daemon must always have the same version. The version of the File Daemon may differ, see chapter about :ref:`backward compatibility <backward-compatibility>`.
```
Comment (bruno-at-bareos) [inline docs/manuals/source/IntroductionAndTutorial/UpdatingBareos.rst:7]: Console and webui should also be updated.

File: docs/manuals/source/IntroductionAndTutorial/UpdatingBareos.rst [hunk 5]
```diff
@@ -14,10 +15,14 @@ If you don’t want to modify the behavior, there is normally no need to modify
...
-they will only be used, when the legacy configuration file does not exist.
+It is generally sufficient, to upgrade to the latest release,
+without having to install any intermediate releases
+(however, it is required to read the release notes of all intermediate releases).
+
+One exception is when using a |mysql| Bareos catalog. The |mysql| backend support has been removed in Bareos :sinceVersion:`21.0.0: MySQL backend removed`. Therefore you first have to upgrade to Bareos 20 and migrate the |mysql| into a |postgresql| Bareos Catalog, see :ref:`section-MigrationMysqlToPostgresql`.
+
+When updating from Bareos < 16.2.4, please read :ref:`section-UpdateToConfigurationSubdirectories`.
 
-See :ref:`section-UpdateToConfigurationSubdirectories` for details and how to migrate to :ref:`section-SubdirectoryConfigurationScheme`.
```
Comment (bruno-at-bareos) [inline docs/manuals/source/IntroductionAndTutorial/UpdatingBareos.rst:20]: We should keep this remark, many many many people have a mixed situation here.

**Review by bruno-at-bareos — COMMENTED** [review]
See differents comments (github suggestion doesn't work today).

File: docs/manuals/source/Appendix/Howtos/RemovingAClient.rst.inc (hunk 1 above)
Comment (joergsteffens) [reply docs/manuals/source/Appendix/Howtos/RemovingAClient.rst.inc:1]: Surely RemovingAClient looks ugly, however we specified to name the File in CamelCase after the first heading, see https://docs.bareos.org/DocumentationStyleGuide/BareosSpecificFormatting/DocumentFileStructure.html

File: docs/manuals/source/Appendix/Howtos/StatisticsCollection.rst.inc (hunk 2 above)
Comment (joergsteffens) [reply docs/manuals/source/Appendix/Howtos/StatisticsCollection.rst.inc:1]: I have not changed in content, I justed splitted it into separate files. Anyhow, I've now adapted the section anchor.

File: docs/manuals/source/IntroductionAndTutorial/InstallingBareos.rst (hunk 3 above)
Comment (joergsteffens) [reply docs/manuals/source/IntroductionAndTutorial/InstallingBareos.rst:320]: What is your point here? Do you propose a change?

File: docs/manuals/source/IntroductionAndTutorial/UpdatingBareos.rst (hunk 5 above)
Comment (joergsteffens) [reply docs/manuals/source/IntroductionAndTutorial/UpdatingBareos.rst:20]: Do you propose to keep the full remark ("With Bareos version 16.2.4 the default  configuration uses ...") ?

File: docs/manuals/source/IntroductionAndTutorial/InstallingBareos.rst (hunk 3 above)
Comment (bruno-at-bareos) [reply docs/manuals/source/IntroductionAndTutorial/InstallingBareos.rst:320]: you already have your **PostgreSQL** database server

File: docs/manuals/source/IntroductionAndTutorial/UpdatingBareos.rst (hunk 5 above)
Comment (bruno-at-bareos) [reply docs/manuals/source/IntroductionAndTutorial/UpdatingBareos.rst:20]: I believe it's nice to keep a cross link to both section, which contain also the previous informations.

File: docs/manuals/source/IntroductionAndTutorial/InstallingBareos.rst (hunk 3 above)
Comment (joergsteffens) [reply docs/manuals/source/IntroductionAndTutorial/InstallingBareos.rst]: Okay, added.

File: docs/manuals/source/IntroductionAndTutorial/UpdatingBareos.rst (hunk 5 above)
Comment (joergsteffens) [reply docs/manuals/source/IntroductionAndTutorial/UpdatingBareos.rst:20]: Readded old explaination.

**Review by bruno-at-bareos — APPROVED** [review]
Reviewed - ready for merge

**Pull request merged**